### PR TITLE
Upgrade dependency (pylint)

### PR DESCRIPTION
**Pull request by aahung**
*Issue #, if available:*

*Why is this change necessary?*

Maintain our dependencies up-to-date.

*How does it address the issue?*

Upgrade the dependencies and adjust our code accordingly.

For pylint, there are new rules such as:
```
samcli/local/lambda_service/local_lambda_invoke_service.py:33:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
samcli/commands/local/start_api/cli.py:117:4: C0415: Import outside toplevel (samcli.commands.exceptions.UserException) (import-outside-toplevel)
samcli/commands/publish/command.py:84:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
```

The rule "import-outside-toplevel" has been disabled since our "violation" looks reasonable to me.

*What side effects does this change have?*

*Did you change a dependency in `requirements/base.txt`?*
    *If so, did you run `make update-reproducible-reqs`* yes

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by CoshUS — APPROVED** [review]

File: samcli/commands/local/lib/swagger/reader.py
Comment (jfuss) [inline samcli/commands/local/lib/swagger/reader.py:9]: Do we need this anymore? Six is only used for py2 to 3 compatibility.

File: samcli/lib/build/app_builder.py
Comment (jfuss) [inline samcli/lib/build/app_builder.py:128]: We need to using the same version of black. I have seen lots of formatting back and forth.

**Review by jfuss — CHANGES_REQUESTED** [review]

File: samcli/lib/build/app_builder.py
Comment (aahung) [reply samcli/lib/build/app_builder.py:128]: I am using `black, version 19.10b0` which is specified here https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md
What version do you have?

File: samcli/commands/local/lib/swagger/reader.py
Comment (aahung) [reply samcli/commands/local/lib/swagger/reader.py:9]: Can we do this in another PR? I tried to remove all `six` usage and found there are a lot of them. This PR is already large in its size, I feel cutting the scope here makes review easier.

File: samcli/lib/build/app_builder.py
Comment (aahung) [reply samcli/lib/build/app_builder.py:128]:
```
(samcli37) ~/P/aws-sam-cli ❯❯❯ black --version                                                                                                                               ✘ 2
black, version 19.10b0
(samcli37) ~/P/aws-sam-cli ❯❯❯ make black
black samcli/* tests/*
All done! ✨ 🍰 ✨
379 files left unchanged.
(samcli37) ~/P/aws-sam-cli ❯❯❯
```

File: samcli/commands/local/lib/swagger/reader.py
Comment (jfuss) [reply samcli/commands/local/lib/swagger/reader.py:9]: Sure. We should probably do this soon though. Looks like six is not in our requirements file, which is concerning.

Comment (aahung) [reply samcli/commands/local/lib/swagger/reader.py:9]: Yeah, I found PyCharm also gave me this warning, looks like some of our dependencies depends on `six`, so my `pip freeze` still shows `six` installed.
I can take on it after getting this one merged.

Comment (aahung) [reply samcli/commands/local/lib/swagger/reader.py:9]: FYI:

```
six==1.15.0
  - astroid==2.4.2 [requires: six~=1.12]
    - pylint==2.3.1 [requires: astroid>=2.2.0,<3]
  - aws-lambda-builders==1.1.0 [requires: six~=1.11]
    - aws-sam-cli==1.6.2 [requires: aws-lambda-builders==1.1.0]
  - aws-sam-translator==1.27.0 [requires: six~=1.11]
    - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
  - docker==4.2.2 [requires: six>=1.4.0]
    - aws-sam-cli==1.6.2 [requires: docker~=4.2.0]
  - jsonschema==3.2.0 [requires: six>=1.11.0]
    - aws-sam-translator==1.27.0 [requires: jsonschema~=3.0]
      - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
  - packaging==20.4 [requires: six]
    - pytest==5.2.1 [requires: packaging]
      - pytest-cov==2.7.1 [requires: pytest>=3.6]
      - pytest-forked==1.1.3 [requires: pytest>=3.1.0]
        - pytest-xdist==1.30.0 [requires: pytest-forked]
      - pytest-rerunfailures==7.0 [requires: pytest>=3.10]
      - pytest-timeout==1.3.3 [requires: pytest>=3.6.0]
      - pytest-xdist==1.30.0 [requires: pytest>=4.4.0]
  - pytest-xdist==1.30.0 [requires: six]
  - python-dateutil==2.8.0 [requires: six>=1.5]
    - arrow==0.16.0 [requires: python-dateutil>=2.7.0]
      - jinja2-time==0.2.0 [requires: arrow]
        - cookiecutter==1.6.0 [requires: jinja2-time>=0.1.0]
          - aws-sam-cli==1.6.2 [requires: cookiecutter~=1.6.0]
    - aws-sam-cli==1.6.2 [requires: python-dateutil~=2.6,<2.8.1]
    - botocore==1.17.63 [requires: python-dateutil>=2.1,<3.0.0]
      - boto3==1.14.63 [requires: botocore>=1.17.63,<1.18.0]
        - aws-sam-cli==1.6.2 [requires: boto3~=1.14.0,>=1.14.23]
        - aws-sam-translator==1.27.0 [requires: boto3~=1.5]
          - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
        - serverlessrepo==0.1.9 [requires: boto3~=1.9,>=1.9.56]
          - aws-sam-cli==1.6.2 [requires: serverlessrepo==0.1.9]
      - s3transfer==0.3.3 [requires: botocore>=1.12.36,<2.0a.0]
        - boto3==1.14.63 [requires: s3transfer>=0.3.0,<0.4.0]
          - aws-sam-cli==1.6.2 [requires: boto3~=1.14.0,>=1.14.23]
          - aws-sam-translator==1.27.0 [requires: boto3~=1.5]
            - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
          - serverlessrepo==0.1.9 [requires: boto3~=1.9,>=1.9.56]
            - aws-sam-cli==1.6.2 [requires: serverlessrepo==0.1.9]
    - dateparser==0.7.6 [requires: python-dateutil]
      - aws-sam-cli==1.6.2 [requires: dateparser~=0.7]
  - serverlessrepo==0.1.9 [requires: six~=1.11]
    - aws-sam-cli==1.6.2 [requires: serverlessrepo==0.1.9]
  - websocket-client==0.57.0 [requires: six]
    - docker==4.2.2 [requires: websocket-client>=0.32.0]
      - aws-sam-cli==1.6.2 [requires: docker~=4.2.0]
```

Comment (jfuss) [reply samcli/commands/local/lib/swagger/reader.py:9]: ya it is coming in as a dependency but if that gets dropped, the cli will explode.

It is not here: https://github.com/aws/aws-sam-cli/blob/develop/requirements/base.txt

Can we add six back in the meantime and then do the follow up with removing six?

Comment (aahung) [reply samcli/commands/local/lib/swagger/reader.py:9]: good idea, done

**Review by jfuss — APPROVED** [review]
LGTM, assuming all tests pass. Thanks for added six back!

**Pull request merged**
